ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/ConnectIntegrationTest.java

```diff
@@ -239,6 +252,68 @@ public void shouldReturnError() {
         containsString("Connector mock-connector already exists"));
   }
 
+  @Test
+  public void shouldReadAndWriteTimestampsToConnect() throws UnsupportedEncodingException {
+    // Given:
+    create("mock-source", ImmutableMap.<String, String> builder()
+        .put("connector.class", "org.apache.kafka.connect.tools.VerifiableSourceConnector")
+        .put("topic", "foo")
+        .put("throughput", "5")
+        .put("id", "123")
+        .put("topic.creation.default.replication.factor", "1")
+        .put("topic.creation.default.partitions", "1")
+        .build());
+    assertThatEventually(
+        () -> ((ConnectorList) ksqlRestClient.makeKsqlRequest("SHOW CONNECTORS;").getResponse()
+            .get(0)).getConnectors().get(0).getState(),
+        is("RUNNING (1/1 tasks RUNNING)"));
+
+    // When:
+    assertThatEventually(
+        () -> ksqlRestClient.makeKsqlRequest("CREATE STREAM FOO (PAYLOAD TIMESTAMP) WITH (KAFKA_TOPIC='foo', VALUE_FORMAT='DELIMITED');")
+            .isSuccessful(), is(true));
+
+    // Then:
+    final RestResponse<List<StreamedRow>> queryFoo = ksqlRestClient.makeQueryRequest("SELECT * FROM FOO EMIT CHANGES LIMIT 1;", 1L);
+    assertThat("successfully queried FOO", queryFoo.isSuccessful());
+    assertThat(queryFoo.getResponse().get(1).getRow().get().getColumns().get(0), is("1970-01-01T00:00:00.000"));
+
+    // When:
+    final RestResponse<KsqlEntityList> barQuery = ksqlRestClient
+        .makeKsqlRequest("CREATE STREAM BAR WITH (value_format='JSON') AS SELECT PAYLOAD FROM FOO;");
```

Comment:
Seems this test is different. Should have have another method for this? Ideally, we should have one test method per test case.